x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -442,7 +440,7 @@ async function _filterAgents(
       runtime_mappings: runtimeFields,
       fields: Object.keys(runtimeFields),
       sort: [{ [sortField]: { order: sortOrder } }],
-      query: { bool: { filter: query } },
+      query: addNamespaceFilteringToQuery({ bool: { filter: [query] } }, currentNameSpace),
```

Comment:
@nchaulet What do you think of this change? I realised that a lot of bulk endpoints were calling `getAgentsById` and then checking `isAgentInNamespace` on each agent. With this, `getAgentsById` directly adds a namespace filter in the ES query. The downside is we lose the more specific "agent not in namespace" error message.

Reply:
it seems okay to me 👍